beancount_ethereum/importer_balances.py

```diff
@@ -0,0 +1,79 @@
+import datetime
+import json
+import os
+
+from beancount.ingest.importer import ImporterProtocol
+from beancount.core.amount import Amount
+from beancount.core.data import Balance, new_metadata
+from beancount.core.number import D
+
+DEFAULT_CURRENCY = "ETH"
+
+
+class Importer(ImporterProtocol):
+    def __init__(
+        self,
+        config_path="config.json",
+    ):
+        with open(config_path, "r") as config_file:
+            self.config = json.load(config_file)
+
+    def name(self) -> str:
+        return "ethereum_balances"
+
+    def identify(self, file) -> bool:
+        name = self.config["name"]
+        return os.path.basename(file.name) == f"{name}-balances.json"
+
+    @property
+    def account_map(self) -> dict:
+        return {key.lower(): value for key, value in self.config["account_map"].items()}
+
+    def account_suffix(self, currency: str) -> str:
+        if "currency_map" in self.config:
+            if currency in self.config["currency_map"]:
+                if "account_suffix" in self.config["currency_map"][currency]:
+                    return self.config["currency_map"][currency]["account_suffix"]
+                else:
+                    return self.config["currency_map"][currency]["commodity"]
+            else:
+                return currency
+        else:
+            return currency
+
+    def commodity(self, currency: str) -> str:
+        if "currency_map" in self.config:
+            if currency in self.config["currency_map"]:
+                return self.config["currency_map"][currency]["commodity"]
+            else:
+                return currency
+        else:
+            return currency
```

Comment:
I would prefer to not duplicate this code.
We can move it to a mixin or just merge everything into single importer.

Mixin:

```python
class ImporterMixin:
    @property
    def account_map(self) -> dict:
        return {key.lower(): value for key, value
                in self.config['account_map'].items()}

...

class Importer(ImporterMixin, ImporterProtocol):
   def __init__(self):

```

Single importer:

```python
beancount_ethereum.importer.Importer(config_path='config.json', import_balances=true),
...

def identify(self, file) -> bool:
    name = self.config["name"]
    return (os.path.basename(file.name) == f'{name}.json' or 
               self.import_balances and 
               os.path.basename(file.name) == f"{name}-balances.json")
```

I think both ways are fine.

Reply:
I think i would me more confortable with a single "all-in-one" importer.
I will modify my PR accordingly.